addons/test_mail_full/models/test_mail_models_mail.py

```diff
@@ -24,27 +24,44 @@ def _compute_access_url(self):
             record.access_url = '/my/test_portal/%s' % self.id
 
 
+class MailTestRatingThread(models.Model):
+    """A model inheriting from mail.thread with minimal fields for testing
+     rating submission without the rating mixin but with the same test code:
+
+     - partner_id:  value returned by the base _rating_get_partner method
+     - user_id: value returned by the base _rating_get_operator method
+     """
+
+    _description = 'Model for testing rating without the rating mixin'
+    _name = 'mail.test.rating.thread'
+    _inherit = 'mail.thread'
+    _order = 'name asc, id asc'
+
+    name = fields.Char('Name')
+    partner_id = fields.Many2one('res.partner', 'Customer')
```

Comment:
I like to use a different field name, because it often highlight issues (like hardcoded 'partner_id' usage, ...). Could we use customer_id on this model as with other models ?

Reply:
I didn’t remembered first but I did it because the base implementation of _rating_get_partner uses partner_id. I changed it to customer_id and added an override of that method to return the customer as the rating partner. But that tells us that without change the template could only be used with model with partner_id (access_token is relying on it when not providing a partner but it is the case in the template). So, indeed, that’s a good idea to not use partner_id in the test to highlight potential problem.